### PR TITLE
We're done. Waiting for review.

### PR DESCRIPTION
**Pull request by Telf92**

### CODE REVIEW

File: src/main/java/se/nackademin/Lemur.java [hunk 1]
```diff
@@ -25,15 +25,15 @@ class Lemur implements Animal {
 
 	
 	public boolean eat() {
-		if (this.hunger < 5) {
+		if (this.hunger > 5) {
```
Comment (ripperdoc) [inline src/main/java/se/nackademin/Lemur.java:28]: Not sure I understand the design intention here. If you eat, and are "hungry" (>5?) you are still hungry after eating even that hunger counter is reset?

(I would probably design is so that isHungry is a method, that compares current hunger with some CONSTANT variable, and returns true if hunger is higher. Then in eat() I would reduce hunger by either a given amount, fixed amount or down to 0)

Comment (ripperdoc) [reply src/main/java/se/nackademin/Lemur.java:28]: Or with eat() you mean shouldEat()? Some documentation at the method would help explain the design intention.

File: src/test/java/se/nackademin/LemurTest.java [hunk 2]
```diff
@@ -13,6 +13,14 @@ public void shouldEat() {
         boolean isHungry = tomislav.eat();
         assertTrue(isHungry);
 
+    }
+    @Test
+    public void shouldNotEat() {
+        Lemur tomislav = new Lemur();
+        tomislav.hunger = 3;
+        boolean isHungry = tomislav.eat();
+        assertFalse(isHungry);
```
Comment (ripperdoc) [inline src/test/java/se/nackademin/LemurTest.java:22]: This is not imported so test doesn't compile or run.

**Comment by ripperdoc** [comment]
Note: I reset the submission to first commit in order to properly show everything that has been added in main.

File: src/main/java/se/nackademin/Lemur.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		else {
+			this.isTired = false;
+		}
+		
+		return this.isTired;
+	}
+	
+	public String snuggle() {
+		if (this.energyLevel < 2) {
+			return "Lemur is too tired to snuggle";
```
Comment (ripperdoc) [inline src/main/java/se/nackademin/Lemur.java:52]: While this is perfectly fine for testing, a suggestion would be to represent the Lemur's current state in an enum variable, e.g. it could either be snuggling, beatboxing or something else. Then a test would ensure that the lemur is in the correct state after some method called.

File: src/test/java/se/nackademin/LemurTest.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
+package se.nackademin;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+public class LemurTest {
+    @Test
+    public void shouldEat() {
```
Comment (ripperdoc) [inline src/test/java/se/nackademin/LemurTest.java:10]: - Your tests shouldEat, shouldNotEat etc contains a number of tests based that depends on some comparison, e.g. that a value should be higher than or lower than a constant. You should document this comparison & constant, best would be with comments in the test. E.g. how do we tell other developers that 3 should be considered not hungry and 8 should be considered hungry, and where the limit is?
- It would make sense to test edge cases, e.g. to test 5 - should that be counted as hungry or not hungry? What about -2 ?

File: src/test/java/se/nackademin/LemurTest.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        assertEquals("Hakkunah Matatatatatatata", response);
+    }
+    @Test
+    public void shouldBackflip() {
+        Lemur lemur = new Lemur();
+        String response = lemur.backflip();
+        assertEquals("Lemur backflips", response);
+    }
+    @Test
+    public void shouldGetOld() {
```
Comment (ripperdoc) [inline src/test/java/se/nackademin/LemurTest.java:53]: This and below tests aren't really testing that getters works, they are more testing that certain defaul values are set, such as age == 1. It's fine to test that default values are set to something specific, but I would name the test more like "defaulAgeShouldBe1" or similar.

File: src/main/java/se/nackademin/Personell.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+	private int salary; // per annum
+	private int balance;
+	
+	public String getName() { return this.name; }
+	public int getAge() { return this.age; }
+	public String getTitle() { return this.title; }
+	public int getSalary() { return this.salary; }
+	public int getBalance() { return this.balance; }
+
+	public boolean sell(boolean isValid) {
```
Comment (ripperdoc) [inline src/main/java/se/nackademin/Personell.java:16]: Not sure on design intention here? "sell" means what for a personell? And what does isValid represent, what is or isn't valid? What's the difference between calling sell(false) and just not calling sell at all?

File: src/main/java/se/nackademin/Personell.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+		if (isValid)
+			this.salary += 1;
+		return isValid;
+	}
+
+	public boolean feedAnimal(Lemur timon) {
```
Comment (ripperdoc) [inline src/main/java/se/nackademin/Personell.java:22]: Should not take a Lemur, but an Animal, in order to use the benefit of the interface (e.g. support feeding any kind of animal.)

File: src/main/java/se/nackademin/Personell.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+	public boolean clean(Enclosure enclosure) {
+		boolean isDirty = enclosure.getDirty();
+		if (isDirty) {
+			enclosure.clean();
+		}
+		return isDirty;
+	}
+
+	public void acceptPayment() {
+		this.balance += this.salary;
```
Comment (ripperdoc) [inline src/main/java/se/nackademin/Personell.java:39]: Sounds like a method that should be tested so that it can't be run too often :)

File: src/main/java/se/nackademin/Personell.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
+package se.nackademin;
+
+class Personell {
+	private String name;
+	private int age;
+	private String title;
+	private int salary; // per annum
+	private int balance;
```
Comment (ripperdoc) [inline src/main/java/se/nackademin/Personell.java:8]: These private instance variables don't have default values, aren't referenced in a constructor and have no setters, so they can never be set to anything.

**Review by ripperdoc — APPROVED** [review]
Good work! Good use of methods that reference other objects, e.g. clean(enclosure) or feedAnimal(lemur). Good amount of tests, the whole team seemed involved. Added a number of comments to think about for next time.
